src/main/java/hudson/plugins/s3/Uploads.java

```diff
@@ -20,7 +21,14 @@ private Uploads() {}
     private final transient HashMap<FilePath, InputStream> openedStreams = new HashMap<>();
 
     public void startUploading(TransferManager manager, FilePath file, InputStream inputsStream, String bucketName, String objectName, ObjectMetadata metadata) throws AmazonServiceException {
-        final Upload upload = manager.upload(bucketName, objectName, inputsStream, metadata);
+        final PutObjectRequest request = new PutObjectRequest(bucketName, objectName, inputsStream, metadata);
+
+        // Set the buffer size (ReadLimit) equal to the multipart upload size,
+        // allowing us to resend data if the connection breaks.
+        request.getRequestClientOptions().setReadLimit(10485760);
```

Comment:
Please, extract a constant variable for this value. And it's not clear why 10485760?